### ISSUE TITLE
学習データのゲーム多様性が汎化性能に与える影響の検証（多ゲーム・少レビュー）

### ISSUE THREAD

**Issue by rindguitar** [rindguitar/game-demand-forecast#30]
## 背景

感情分析フェーズの内容でLTを行った際、以下の指摘を受けた。

> なぜ7ゲーム・10000件のレビューで学習させたのか？ 多ゲーム・少レビューで学習させてみては？

現状の学習データは `scripts/collect/collect_dataset_10k.py` の通り、**7タイトル固定**で各約1428件を収集している。

| タイトル | リリース年 |
|---|---|
| Baldur's Gate 3 | 2023 |
| Cyberpunk 2077 | 2020 |
| Stardew Valley | 2016 |
| The Witcher 3 | 2015 |
| Grand Theft Auto V | 2013 |
| Counter-Strike 2 | 2023 |
| Dota 2 | 2013 |

7タイトルにした理由は「1タイトルあたりの件数を確保するため」だったが、**ゲームの種類を増やす方向は検討していなかった**。

## 何が問題になりうるか

モデルが学習しているのが「感情の表現」なのか、「この7タイトル特有の言い回し」なのかが区別できていない可能性がある。

たとえば Dota 2 や CS2 のレビューには MOBA/FPS 特有の語彙（`smurf`, `throw`, `griefing` 等）が頻出する。モデルがこうした語を「ネガティブの目印」として覚えてしまっていても、7タイトルの中だけで評価している限り高いスコアが出てしまう。実運用では未知のタイトルのレビューを分類することになるため、そこで精度が落ちる恐れがある。

### 現状の評価では、この問題を検出できない

`scripts/nlp/train_sentiment.py` の L105-L117 でのデータ分割は `stratify=df['label']` のみを指定したランダム分割になっている。

```python
train_df, temp_df = train_test_split(
    df,
    test_size=(val_ratio + test_ratio),
    random_state=seed,
    stratify=df['label']
)
```

つまり **同じ7タイトルのレビューが train にも test にも混在している**。現状の Test Acc 84.66% は「学習で見たタイトルに対する精度」であり、未知のタイトルに対する精度ではない。

ここが本issueの中心。**ゲーム数を増やすかどうかの前に、まず「未知のタイトルで測る評価」を用意しないと、指摘への回答にならない。** 50タイトルで学習しても、その50タイトルからランダム分割してテストしていては、何も検証したことにならない。

## 検証したいこと

学習データに含まれるゲームの種類を増やすと、**学習に一度も出てこなかったタイトル**に対する精度は上がるのか。

### 1. 何を固定して、何を動かすか

総レビュー件数を10000件に固定したままゲーム数だけ変えるのか、それとも件数も一緒に増やすのか。前者なら「多様性の効果」だけを切り出せるが、後者と混ぜると原因が分からなくなる。

- 案A: 総件数10000件で固定し、ゲーム数のみ変える（7 / 20 / 50 など）
- 案B: 1タイトルあたりの件数を固定し、ゲーム数を増やす（総件数も増える）

### 2. 評価用データをどう作るか

上記の通り、ここが結論を左右する。

- 学習に使わないタイトルを何本か確保し、そこから評価用データを作る（タイトル単位で分ける）
- 除外するタイトルの選び方も要検討。ジャンル・リリース年・レビュー総数が学習側と偏らないようにする必要がある
- 既存の10000件モデルと比較するには、**同じ評価用データで両方を測り直す**必要がある（現状の84.66%とは直接比較できない）

### 3. ゲーム数の水準をどう置くか

何本ずつ試せば傾向が見えるか。刻みが細かすぎると学習回数がかさむ（GTX 1060・6GBの制約あり）。

### 4. 収集コストと現実的な制約

- Steam APIへのリクエスト数がタイトル数に比例して増える
- **balanced収集が成立しないタイトルが出る**。`collect_balanced_reviews()` は positive/negative を同数集める前提だが、評価の高いインディータイトルではネガティブレビューが目標件数に届かない可能性がある。件数が揃わない場合の扱い（そのタイトルを捨てる／件数を減らして揃える）を決める必要がある
- タイトルの選定基準（レビュー総数の下限、ジャンルの散らし方、リリース年の分布）

## 完了条件

- 未知タイトルに対する精度が測れる評価用データが用意されている
- ゲーム数の異なる条件で学習・評価した結果が比較できる形で残っている
- 「7タイトルで良かったのか」に対して、数字を添えて答えられる状態になっている

## 関連

- #19 感情分析モデルの高精度化（目標: 平均Test Acc 90%）
  - #19は皮肉・否定構文といった**表現の難しさ**への対応。本issueは**学習データの構成**の問題であり、目的が異なる。ただし本issueで未知タイトルへの評価基盤ができれば、#19の効果測定にも使える
- `scripts/collect/collect_dataset_10k.py`（現状の収集スクリプト）
- `scripts/nlp/train_sentiment.py`（現状のデータ分割）
- 時系列予測フェーズと並行、または完了後に着手予定
